src/nautilus/paging.c

```diff
@@ -477,12 +477,16 @@ __construct_tables_4k (pml4e_t * pml, ulong_t bytes)
     for (i = 0; i < NUM_PML4_ENTRIES && filled_pdpts < num_pdpts; i++) {
 
         pdpte_t * pdpt = (pdpte_t*)PTE_ADDR(pml[i]);
-        __fill_pdpt(pdpt, PS_4K, addr, num_pds, PTE_PRESENT_BIT | PTE_WRITABLE_BIT);
+        ulong_t pdpt_to_fill = ((num_pds - filled_pds) > NUM_PDPT_ENTRIES) ? NUM_PDPT_ENTRIES:
```

Comment:
The spirit of this line is correct, but it should be `pd_to_fill`

Reply:
Ah yes, pdpte name is better